apps/extension/src/storage/onboard.ts

```diff
@@ -0,0 +1,51 @@
+import { ExtensionStorage } from '@penumbra-zone/storage/chrome/base';
+import { localExtStorage } from '@penumbra-zone/storage/chrome/local';
+import { SessionStorageState } from '@penumbra-zone/storage/chrome/session';
+import { LocalStorageState } from '@penumbra-zone/storage/chrome/types';
+
+export const waitForStorageKey = async <
+  S extends LocalStorageState | SessionStorageState,
+  K extends keyof S,
+  I extends keyof S[K],
+>(
+  storage: ExtensionStorage<S>,
+  storageKey: K & string,
+  innerKey?: I,
+): Promise<I extends undefined ? S[K] : S[K][I]> => {
+  type Stored = I extends undefined ? S[K] : S[K][I];
+  const { promise, resolve, reject } = Promise.withResolvers<Stored>();
+
+  const storedObject: S[K] = await storage.get(storageKey);
+  const storedValue = (innerKey != null ? storedObject[innerKey] : storedObject) as
```

Comment:
Can we convert all equality checks to be strict equality pls? It makes it harder to discern what the type coercision is. In general, folks in the industry try not to use it.

Also, shouldn't these be undefined checks? aka `innerKey !== undefined`